samples/config_helper_vulkan.cc

```diff
@@ -757,14 +791,33 @@ amber::Result ConfigHelperVulkan::CreateDeviceWithFeatures1(
 amber::Result ConfigHelperVulkan::CreateDeviceWithFeatures2(
     const std::vector<std::string>& required_features,
     VkDeviceCreateInfo* info) {
+  variable_pointers_feature_.sType =
+      VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VARIABLE_POINTER_FEATURES_KHR;
+  variable_pointers_feature_.pNext = nullptr;
+
+  available_features2_.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2_KHR;
+  available_features2_.pNext = &variable_pointers_feature_;
+
+  std::vector<std::string> feature1_names;
+  for (const auto& feature : required_features) {
+    // No dot means this is a features1 feature.
+    if (feature.find_first_of('.') == std::string::npos) {
+      feature1_names.push_back(feature);
+      continue;
+    }
+
+    if (feature == kVariablePointers)
+      variable_pointers_feature_.variablePointers = VK_TRUE;
+    else if (feature == kVariablePointersStorageBuffer)
+      variable_pointers_feature_.variablePointersStorageBuffer = VK_TRUE;
```

Comment:
Comment from @alan-baker:

"The spec says if you turn on VK_KHR_variable_pointers you MUST support variablePointersStorageBuffer so I'd default that to true if the extension or VariablePointerFeatures.variablePointers is mentioned"

Do we want the implicit setting to true, or should we make the script have to be explicit?

Reply:
I'm ok with forcing the script to be explicit.  I think.

Reply:
What about VariablePointerFeatures.variablePointers implying both variables?

Reply:
I'm leaning towards explicit so yo have to specify both. It makes the test a lot clearer and removes spooky action at a distance.